### PR TITLE
#5683: don't generalize if we're already generalizing

### LINKED ISSUE #5683: let: unexpected generalization of variable
*Agda 2.6.2.0.20211129*

When using generalized variables in a constructor, I expect to get the same behaviour as if I had explicitly introduced it with a ∀ quantifier.
This breaks by the following MWE:
```agda
postulate
  X : Set
  ∀≡ : ∀ (x : X) → x ≡ x

variable  x : X

data C : Σ X (λ x → x ≡ x) → Set where
  _ :
    -- works if I remove this comment
    -- ∀ {x} → 
    let
      eq : x ≡ x
      eq = ∀≡ x
    in
    C (x , eq)
```
```
x != x of type X
(because one has de Bruijn index 1 and the other 0)
when checking that the expression ∀≡ x has type x ≡ x
```
It seems that the `let`-clause does its own generalization (producing `x = # 0`) and shadows the previous `x = # 1`.

### PR DESCRIPTION
**Pull request by UlfNorell**
Fixes #5683

### CODE REVIEW

File: test/Succeed/Issue5683.agda [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  X  : Set
+  ∀≡ : ∀ (x : X) → x ≡ x
+
+variable x : X
+
+data C : Σ X (λ x → x ≡ x) → Set where
+  mkC :
+    let
+      eq : x ≡ x  -- don't generalize over x at eq
+      eq = ∀≡ x
```
Comment (andreasabel) [inline test/Succeed/Issue5683.agda:15]: Why not `refl`?

**Review by andreasabel — APPROVED** [review]

File: test/Succeed/Issue5683.agda (hunk 1 above)
Comment (UlfNorell) [reply test/Succeed/Issue5683.agda:15]: Why indeed? I copied the test case from the issue report.

Comment (andreasabel) [reply test/Succeed/Issue5683.agda:15]: _Kopieren geht über studieren._

Comment (UlfNorell) [reply test/Succeed/Issue5683.agda:15]: My German is a bit non-existing, but I think that means "copying is superior to studying".

Comment (omelkonian) [reply test/Succeed/Issue5683.agda:15]: Well, the original issue would not arise with `refl`, it's an artificial example to trigger the unexpected generalisation.

Comment (andreasabel) [reply test/Succeed/Issue5683.agda:15]: I see, thanks!

Comment (omelkonian) [reply test/Succeed/Issue5683.agda:15]: However, it can be simplified to `refl {x = x}` without the need of any postulates, I will submit a PR.

**Pull request merged**